regex/rvalidation/ipv4Validation.go

```diff
@@ -0,0 +1,12 @@
+package rvalidation
+
+import "regexp"
+
+// ValidateIPv4 validates a string as an ipv4 address.
+func ValidateIPv4(hash string) bool {
+	pattern := `^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$`
```

Comment:
@nullawhale if possible can you please split pattern in sub patterns, it will be easy for others to understand.
Example
Ip :   a.b.c.d
pattern = patternA+patternB+patternC+patternD